File: examples/tsj38-kotlin-parity/kotlin-app/src/main/kotlin/dev/tsj/reference/repository/InMemoryOrderRepository.kt
package dev.tsj.reference.repository

import dev.tsj.reference.domain.Order

class InMemoryOrderRepository : OrderRepository {
    private val orders = mutableListOf<Order>()
    private var nextId = 1

    override fun list(): List<Order> = orders.toList()

    override fun save(customer: String, total: Double): Order {
        val order = Order("order-$nextId", customer, total)
        nextId += 1
        orders.add(order)
        return order
    }
}

File: examples/tsj38-kotlin-parity/ts-app/src/repository/in-memory-order-repository.ts
import type { Order } from "../domain/order";
import type { OrderRepository } from "./order-repository";

export class InMemoryOrderRepository implements OrderRepository {
  private readonly orders: Order[];
  private nextId: number;

  constructor() {
    this.orders = [];
    this.nextId = 1;
  }

  list(): Order[] {
    return this.orders.slice();
  }

  save(customer: string, total: number): Order {
    const order: Order = {
      id: "order-" + this.nextId,
      customer: customer,
      total: total
    };
    this.nextId = this.nextId + 1;
    this.orders.push(order);
    return order;
  }
}
